### PR TITLE
RNNT Loss remove null grads

### PR DESCRIPTION
**Pull request by carolineechen**
Previously, forward calls to `RNNTLoss` or `rnnt_loss` would fail with a `cannot call get_autograd_meta() on undefined tensor` error when `requires_grad=False`. This PR resolves this issue so that forward calls can be made even when the logits do not have `requires_grad=True`.

### CODE REVIEW

File: torchaudio/csrc/rnnt/cpu/compute.cpp [hunk 1]
```diff
@@ -87,10 +87,7 @@ std::tuple<torch::Tensor, c10::optional<torch::Tensor>> compute(
   torch::Tensor costs = torch::empty(
       options.batchSize_ * options.nHypos_,
       torch::TensorOptions().device(logits.device()).dtype(logits.dtype()));
-  c10::optional<torch::Tensor> gradients = c10::nullopt;
-  if (logits.requires_grad()) {
-    gradients = torch::zeros_like(logits);
-  }
+  c10::optional<torch::Tensor> gradients = torch::zeros_like(logits);
```
Comment (mthrok) [inline torchaudio/csrc/rnnt/cpu/compute.cpp:90]: Here, Tensor with the same number of elements as `logits` is initialized, even when `requires_grad=False`.

Looking at all the cases, the storage of the tensor is passed to `Compute` function.
Can we handle  the case `c10::optional<torch::Tensor> gradients == nullopt` in `Compute` function, ideally, without allocating `gradient` tensor?

Comment (mthrok) [reply torchaudio/csrc/rnnt/cpu/compute.cpp:90]: It is eventually passed to https://github.com/pytorch/audio/blob/8a347b62cf5c907d2676bdc983354834e500a282/torchaudio/csrc/rnnt/cpu/cpu_kernels.h#L373

and it seems to me not easy to fix this function. So I think the current approach of quick fix is good enough. Maybe we can file the issue for the further improvement (handle `gradient==nullptr` in `ComputeGradient`) in the future.

**Review by mthrok — APPROVED** [review]

**Comment by mthrok** [comment]
@carolineechen Could we have a simple test to make sure that passing Tensor with `requires_grad=False` to RNN-T loss works?

File: test/torchaudio_unittest/rnnt/rnnt_loss_impl.py [hunk 2]
```diff
@@ -27,6 +27,12 @@ def test_basic_backward(self):
         loss = rnnt_loss(logits, targets, logit_lengths, target_lengths)
         loss.backward()
 
+    def test_basic_forward_no_grad(self):
```
Comment (mthrok) [inline test/torchaudio_unittest/rnnt/rnnt_loss_impl.py:30]: Can we give the description of the context?
I think for future maintainer, this test looks insignificant so it's better to give explanation why this test matters.
Something like

```
In early stage, `RNNTLoss` caused segmentation fault when `logits` have `requires_grad = False`.
This test makes sure that this segmentation fault is caught properly.

See also https://github.com/pytorch/audio/pull/1707
```

Comment (carolineechen) [reply test/torchaudio_unittest/rnnt/rnnt_loss_impl.py:30]: I've added a comment in the following PR [here](https://github.com/pytorch/audio/pull/1711/commits/59366cc616f4adced46c3eed11367023e5cbc0d3#diff-cf61df2ded7de3af992d120c6b1530bddb6333ad62e825ce4c9251dafe9afd1dR459)
